src/utils/getAreasList/getAreasList.js

```diff
@@ -21,26 +20,33 @@ export type TAreasList = {
 }
 
 export default function getAreasList(props: TProps): TAreasList {
-  const areas = Object.keys(props).reduce(
+  const propKeys = Object.keys(props)
+
+  /*
+    if the propValue is not a string it is disqualified from being a template
+    from the very get go, mini-optimization there + it removes the need to
+    type-cast the value passed to sanitizeTemplateString since flow can now
+    infer that the value can only be a string otherwise it would have been
+    filtered out already
+  */
+  const templatePropKeys = propKeys.filter(
+    (propKey) =>
+      typeof props[propKey] === 'string' && /template/i.test(propKey),
+  )
+
+  const areasList = templatePropKeys.reduce(
     (res, propName) => {
-      const { purePropName, breakpointName, behavior } = parsePropName(propName)
-      const isTemplateProp = purePropName === 'template'
-      const propValue =
-        isTemplateProp &&
-        sanitizeTemplateString(((props[propName]: any): string))
-
-      const nextAreas =
-        isTemplateProp && propValue ? res.areas.concat(propValue) : res.areas
-      const nextTemplates = isTemplateProp
-        ? res.templates.concat({
-            breakpoint: Layout.getBreakpoint(breakpointName),
-            behavior,
-            areas: propValue,
-          })
-        : res.templates
+      const { breakpointName, behavior } = parsePropName(propName)
+      const propValue = sanitizeTemplateString(props[propName])
+      const nextAreas = propValue ? res.areas.concat(propValue) : res.areas
+      const nextTemplates = res.templates.concat({
+        breakpoint: Layout.getBreakpoint(breakpointName),
+        behavior,
+        areas: propValue,
+      })
 
       return {
-        areas: Array.from(new Set(nextAreas)),
```

Comment:
So you remove set construction here to save some resources as opposed to making it a set at the end of the function call, is that correct?

Reply:
We still need to make the Array -> Set conversion but we really only care about it when it returns from the function, doing it on every reduction is not necessary.